### PR TITLE
DOC: Add a contribution guide

### PR DESCRIPTION
**Pull request by BenjaminBossan**
As discussed internally, we would like to add a contribution guide to facilitate contributions from the community and clarify the requirements.

While looking at the README.md, I wondered if we should remove the "backlog" section, as it looks outdated.

### CODE REVIEW

**Review by pacman100 — APPROVED** [review]
Thank you @BenjaminBossan for the detailed contributor guide, LGTM! Let's remove the Backlog section.

**Review by younesbelkada — APPROVED** [review]
Thank you!

File: docs/source/developer_guides/contributing.mdx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+Once you have something that seems to be working, don’t hesitate to create a draft PR, even if it’s not in a mergeable state yet. The maintainers will be happy to give you feedback and guidance along the way.
+
+## Adding other features
+
+It is best if you first open an issue on GitHub with a proposal to add the new feature. That way, you can discuss with the maintainers if it makes sense to add the feature before spending too much time on implementing it.
+
+New features should generally be accompanied by tests and documentation or examples. Without the latter, users will have a hard time discovering your cool new feature.
+
+Changes to the code should be implemented in a backwards compatible way, i.e. existing code should continue to work the same way after the feature is merged.
```
Comment (stevhliu) [inline docs/source/developer_guides/contributing.mdx:90]:
```suggestion
Changes to the code should be implemented in a backward-compatible way. For example, existing code should continue to work the same way after the feature is merged.
```

**Review by stevhliu — APPROVED** [review]
Really nice and concise! 👍

**Comment by BenjaminBossan** [comment]
@stevhliu Thanks for the suggestions, I applied them. One learning for me is not to break sentences with code snippets :)

**Pull request merged**